docs/proposals/08-shoot-additional-container-runtimes.md

```diff
@@ -0,0 +1,139 @@
+
+## Design Details
+
+1. Change the nodes container runtime to work with CRI and containerd by default.
+    1. In order to configure each worker machine in the cluster to work with CRI, the following configurations should be done:
+        1. Add kubelet execution flags:
+            1. --container-runtime=remote
+            2. --container-runtime-endpoint=unix:///.../containerd.socket
+        2. Make sure that containerd configuration file exist in path /etc/containerd/config.toml and it is configured to enable the CRI plugin (In the containerd config.toml "cri" plugin must not be disabled).
+        
+    2. Containerd socket path and containerd configurations are different for each OS. To make sure the configurations above are set in each worker machine, each OS extension would be responsible to configure them during the reconciliation of the 
+       OperatingSystemConfig when it creates the cloud init.
+        1. os-ubuntu - 
+            1. Add the following flags to the kubelet execution command:
+                1. --container-runtime=remote
+                2. --container-runtime-endpoint=unix:///run/containerd/containerd.sock
+            2. Create a containerd configuration file: /etc/containerd/config.toml based on the current default configuration (containerd config default).
+        2. os-coreos - 
+            1. Add the following flags to the kubelet execution command:
+               1. --container-runtime=remote
+               2. --container-runtime-endpoint=unix:///var/run/docker/libcontainerd/docker-containerd.socket
+            2. Create a containerd configuration file: /etc/containerd/config.toml based on the file in the path: __TBD__ and remove the CRI disabled plugin. 
+            3. Override the default containerd.service unit. Create a new unit in: /etc/systemd/system/containerd.service unit that will start the containerd service with a configuration file path: /etc/containerd/config.toml.
+        3. os-suse-jeos - __TBD__     
+        
+    3. Docker-monitor daemon and rotate log should be replaced with equivalent conatinerd services. __TBD__ 
+
+2. Validate workers additional runtime configurations.  
+3. Add support for each additional container runtime in the cluster.   
+    1. In order to install each additional available runtime in the cluster we should:
+        1. Install the runtime binaries in each Worker's pool nodes that specified the runtime support.
+        2. Apply the relevant RuntimeClass to the cluster.
+        
+    2. The installation above should be done by a new kind of extension: RuntimeContainer resource. For each runtime container type (Kata-container/gvisor) a dedicate extension controller will be created. 
```

Comment:
Since the actual functionality is provided by installing a chart into the target shoot, is it really necessary to introduce a new type of Gardener Extension here? Maybe there's already an extension type that supports installing a Chart into a Shoot?

Reply:
We have `Extension` CRDs that can do arbitrary things (including installing charts into shoots), but for this type of extension (container runtimes) I think it makes sense to make this explicit and introduce a new `ContainerRuntime` kind.

Reply:
See above, combination of OS and container runtime and invention of a contract that is not sufficient.

Reply:
Example of implementation for Ubuntu and gVisor:
https://github.com/nimrodoron/gardener/pull/9/files
https://github.com/nimrodoron/gardener-extensions/pull/10/files